OmsAgent/omsagent.py

```diff
@@ -369,7 +369,7 @@ def stop_telemetry_process():
             for pid in f.readlines():
                 # Verify the pid actually belongs to omsagent.
                 cmd_file = os.path.join("/proc", str(pid.strip("\n")), "cmdline")
-                if os.path.exists(cmd_file):
+                if os.path.exists(cmd_file) and (os.path.getsize(cmd_file) > 0):
```

Comment:
This is a time of check to time of use error waiting to happen.

Between this line and the next the PID may be gone, and the `open` below will fail. Instead the code should attempt to open the file unconditionally, and have a try/catch surrounding it for failure due to the PID not existing.

That solves one issue...